apis/r/tools/get_tarballs.R

```diff
@@ -24,15 +18,23 @@ if (isMac) {
     stop("Unsupported platform for downloading artifacts. Please have TileDB Core installed locally.")
 }
 
+if (any(Sys.which(c("cmake", "git")) == ""))
+    stop("The 'cmake' and 'git' commands are required.")
+
 tarball <- "tiledb.tar.gz"
 if (!file.exists(tarball)) download.file(url, tarball, quiet=TRUE)
-if (!dir.exists("tiledb")) untar(tarball, exdir="tiledb")
 if (!dir.exists("inst/tiledb")) untar(tarball, exdir="inst/tiledb")
-archincl <- paste0("-I", system.file("include", package="arch"))
+
+soma_url <- "https://github.com/single-cell-data/TileDB-SOMA/archive/refs/tags/1.2.3.tar.gz"
+soma_tarball <- "tiledbsoma.tar.gz"
+if (!file.exists(soma_tarball)) download.file(soma_url, soma_tarball, quiet=TRUE)
+if (!dir.exists("inst/tiledbsoma")) untar(soma_tarball, exdir="inst/tiledbsoma")
+cat("** building libtiledbsoma\n")
+system("tools/build_libtiledbsoma.sh")
 
 mkvar <- readLines("src/Makevars.in")
 mkvar <- gsub("@cxx17_macos@", macosver, mkvar)
-mkvar <- gsub("@tiledb_include@", paste("-I../inst/tiledb/include", archincl), mkvar)
-mkvar <- gsub("@tiledb_libs@", "-ltiledb -L../inst/tiledb/lib", mkvar)
-mkvar <- gsub("@tiledb_rpath@", "-Wl,-rpath,'$$ORIGIN/../tiledb/lib'", mkvar)
+mkvar <- gsub("@tiledb_include@", "-I../inst/tiledb/include -I../inst/tiledbsoma/include", mkvar)
+mkvar <- gsub("@tiledb_libs@", "-ltiledb -L../inst/tiledb/lib -ltiledbsoma -L../inst/tiledbsoma/lib", mkvar)
+mkvar <- gsub("@tiledb_rpath@", "-Wl,-rpath,'$$ORIGIN/../tiledb/lib' -Wl,-rpath,'$$ORIGIN/../tiledbsoma/lib'", mkvar)
```

Comment:
Do we need to check for `lib` vs `lib64` vs `lib/x86_64-linux-gnu` usage?

Reply:
AAGICT "apparently not."  The first part (ending in `../tiledb/lib` is what we used for years for Core artifact.  The lib64 that it brings along is seemingly just noise.  What matters is what we copy in there as `liledb/lib/libtiledb.so`.  

So that part is 'reduce-reuse-recycle' which I adopted for the (new here) `libtiledbsoma.so` part.  And ... it works on macOS and Linux in CI.